### DISCUSSION TITLE
[Suggestion] Selecting a battery charge limit higher than 80% for devices that only respect 80% charge limit.

### DISCUSSION THREAD

**Discussion by GTMoraes** [seerge/g-helper#3193] [Ideas]
Hey,

I used to use a rather cumbersome script to do this, and it'd be great if G-Helper did it natively. 

Basically, if the device doesn't respect any value higher than 80% for charging limit, you can wait until it reaches the desired charge % and then select the charge limit to 80%. This way, it'll stop charging at that percentage and feed on AC.

So my suggestion's to have G-helper automatically do this switch. There could be a checkbox to mark that your laptop only respects 80% charge limit, and then G-Helper would monitor the current charge and then apply this trick whenever the charge is at or over the specified value. If it's below the desired charge, it just removes the charge limit until it reaches the desired charge again.

Drawbacks:
- If G-Helper is off and the device begins charging below the charge limit (i.e. when turned off or sleeping), it'll charge to 100%.
- If G-Helper is off, was at the desired charge limit and discharged below 80% (i.e. when turned off or sleeping), it'll only charge to 80%.

**Comment by seerge** [comment]
@GTMoraes hello, i'm not a big fan of reinventing the wheel here honestly.

First of all some models (literally mine G14 2022) can discharge, when you set lower limit and still keep device plugged in. So it will result in limitless charge-discharge mini cycles.

Also it's quite hectic logic, will need constant charge monitoring.

And finally - what is the actual practical purpose ? To preserve battery you really need to set 80% or lower. If you want to charge device to full before long day - just click existing one-time `100%` button.

**Reply by GTMoraes** [reply]
That's cool, no problem! I didn't know other models would discharge to the specified value. That's actually a desirable feature, if you think about it.

The practical purpose, at least for me, is that I don't kinda like the last 10% of charge, and my battery's ill suited for this laptop. 
It lasts roughly an hour and a half when on battery, from 100% to 7%, and giving up 20% hurts; while giving up 10% I found as a reasonable tradeoff between full charge degradation and uptime on battery.

I also did a similar script for my Lenovo Yoga laptop, because even though it has a good battery life of over 6 hours, the battery conservation mode is hard set to 55%. So I ran that script to hold it back at 85%.    

Because of that, I thought most laptops with that hardcoded battery limit could enjoy a "freely selectable percentage", but I understand now that it isn't as clear cut as I thought.
